### PR TITLE
AdsNative adapter for AMP

### PR DESCRIPTION
**Pull request by sammerry**
Adding suport for [AdsNative](http://adsnative.com/) native advertising amp-ad element

## Example usecase:
```
<amp-ad width=300 height=250
      type="adsnative"
      data-anapiid="t8AjW-y8rudFGzKft_YQcBZG1-aGZ5otj5QdsKaP"
      data-ancat="IAB1,IAB2"
      data-antid="abc"
      data-ankv="key:val,key:val2">
</amp-ad>
```

**Required**
- width:        required by amp
- height:       required by amp
- data-anapiid: the api id may be used instead of network and widget id
- data-annid:   the network id must be paired with widget id
- data-anwid:   the widget id must be paired with network id

**Optional**
- data-anapiid: the api id
- data-anwid:   the widget id
- data-antid:   the template id
- data-ancat:   a comma separated list of categories
- data-ankv:    a list of key value pairs in the format "key1:value1, key2:value2"

### CODE REVIEW

**Comment by sammerry** [comment]
signed the cla

**Comment by sammerry** [comment]
signed again with work email

**Comment by jridgewell** [comment]
You might have to say (explicitly) "I signed it!".

**Comment by sammerry** [comment]
I signed it!

File: ads/adsnative.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  // convert string to array
+  let actualcat = undefined;
+  if (data.ancat) {
+    actualcat = data.ancat.split(',');
+  }
+
+  // populate settings
+  const gan = global._AdsNativeOpts = {};
+  gan.apiKey = data.anapiid;
```
Comment (lannka) [inline ads/adsnative.js]: how about assign the valuable in one statement:

```
const gan = global._AdsNativeOpts = {
  apiKey: data.anapiid,
  networkKey: data.annid,
  ...
};
```

**Comment by lannka** [comment]
@sammerry if you signed as a corp, please also mention the exact corp name.

File: ads/adsnative.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    actualkv = {};
+    const arraykv = data.ankv.split(',');
+    for (const k in arraykv) {
+      const kv = arraykv[k].split(':');
+      actualkv[kv.pop()] = kv.pop();
+    }
+  }
+
+  // convert string to array
+  let actualcat = undefined;
```
Comment (lannka) [inline ads/adsnative.js]:
```
const actualcat = data.ancat ? data.ancat.split(',') : undefined;
```

**Comment by sammerry** [comment]
@lannka I signed as an individual from both my personal and work addresses.

**Comment by lannka** [comment]
> If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address. Check your existing CLA data and verify that your email is set on your git commits.

@sammerry Probably you've already did this, but just to confirm. The CLA check should happen automatically by the bot.

**Comment by sammerry** [comment]
It was the email attached to the commit. Updated and everything looks fine.

**Comment by lannka** [comment]
Not sure what woke up the CLA robot, but yeah, it passed!

One last thing before I can merge, please resolve merge conflict. Sorry about that, a few contributors are changing the file meanwhile :-).

**Comment by sammerry** [comment]
@lannka rebased against latest master

**Pull request merged**